doc/developer_guide/psyir.rst

```diff
@@ -490,6 +490,42 @@ var2, var3)` would be represented by a
 The PSyIR supports the concept of named arguments for operation
 nodes, see the :ref:`named_arguments-label` section for more details.
 
+IntrinsicCall Nodes
+-------------------
+
+There are certain intrinsic functions that do not lend themselves to
+being represented as `Operation` nodes. For example, Fortran's
+`allocate` statement has various *optional* arguments, one of which
+(`stat`) may be used to store a return value. It's therefore not clear
+what an allocation 'operation' would assign to (an `Operation`
+must be a child of a `Statement` and therefore could not be included
+in a `Schedule` on its own). Similarly, Fortran's `MAXVAL` and
+`MINVAL` intrinsics have optional `dim` and `mask` arguments. In order
+to represent these using `Operation` nodes, we would one for each
+of the four possible forms of each intrinsic.
+
+Therefore, to support intrinsic 'operations' that have optional
+arguments, the PSyIR has the :ref_guide:`IntrinsicCall psyclone.psyir.nodes.html#psyclone.psyir.nodes.IntrinsicCall` Node. This single class
```

Comment:
Is it possible to break this line a bit? Maybe before and after the link.

Reply:
Yes, done.